packages/nimble-components/src/banner/styles.ts

```diff
@@ -0,0 +1,197 @@
+import { css } from '@microsoft/fast-element';
+import { display } from '@microsoft/fast-foundation';
+import {
+    BannerFail100DarkUi,
+    Black75,
+    Fail100LightUi,
+    Information100DarkUi,
+    Information100LightUi,
+    Warning100DarkUi,
+    Warning100LightUi,
+    White
+} from '@ni/nimble-tokens/dist/styledictionary/js/tokens';
+
+import {
+    actionRgbPartialColor,
+    applicationBackgroundColor,
+    bodyFont,
+    borderHoverColor,
+    buttonLabelFontColor,
+    controlHeight,
+    controlSlimHeight,
+    fillSelectedColor,
+    iconColor,
+    iconSize,
+    linkActiveFontColor,
+    linkDisabledFontColor,
+    linkFontColor,
+    standardPadding
+} from '../theme-provider/design-tokens';
+import { Theme } from '../theme-provider/types';
+import { hexToRgbaCssColor } from '../utilities/style/colors';
+import { themeBehavior } from '../utilities/style/theme';
+
+export const styles = css`
+    ${display('flex')}
+
+    :host {
+        width: 100%;
+        font: ${bodyFont};
+        font-size: 12.8px;
+        align-items: top;
+        overflow: hidden;
+        color: ${White};
+        ${iconColor.cssCustomProperty}: ${hexToRgbaCssColor(White, 0.6)};
+    }
+
+    :host(:not([open])) {
+        display: none;
+    }
+
+    .container {
+        display: flex;
+        width: 100%;
+    }
+
+    .icon {
+        width: 48px;
+        display: flex;
+        justify-content: center;
+        margin-top: 8px;
+        flex: 0 0 auto;
+    }
+
+    .text {
+        display: inline;
+        margin-top: 7px;
+        margin-bottom: 7px;
+    }
+
+    ::slotted([slot='title']) {
+        font-weight: bold;
+        padding-right: 8px;
+        white-space: nowrap;
+    }
+
+    .controls {
+        margin-left: auto;
+        display: flex;
+        justify-content: center;
+    }
+
+    slot[name='action'] {
+        display: flex;
+        align-content: center;
+    }
+
+    ::slotted(nimble-anchor[slot='action']) {
+        ${linkFontColor.cssCustomProperty}: ${White};
+        ${linkDisabledFontColor.cssCustomProperty}: ${White};
+        ${linkActiveFontColor.cssCustomProperty}: ${hexToRgbaCssColor(
+            White,
+            0.6
+        )};
+        font-size: 12.8px;
+        white-space: nowrap;
+        margin-top: 7px;
+        margin-left: ${standardPadding};
+    }
+
+    ::slotted(nimble-button[slot='action']) {
+        height: ${controlSlimHeight};
+        ${buttonLabelFontColor.cssCustomProperty}: ${White};
+        ${fillSelectedColor.cssCustomProperty}: ${hexToRgbaCssColor(
+            White,
+            0.2
+        )};
+        ${borderHoverColor.cssCustomProperty}: ${White};
+        font-size: 12.8px;
+        white-space: nowrap;
+        margin-top: calc((${controlHeight} - ${controlSlimHeight}) / 2);
+        margin-left: ${standardPadding};
```

Comment:
The styles of `.controls`, `.dismiss`, `.dismiss nimble-button` and the 3 `slot='action` styles seem overly complicated (though I realize some amount of complication is needed to get the colors to all be correct). I'm mostly thinking that the layout of everything within `.controls` is complicated.

Could you do some combination of the following:
- provide a `margin-right` and a `gap` for `.controls` and completely remove the `<div class="dismiss">` from the template
- if you set the height to be 32px (or the appropriate token) on `.controls` and then center the contents vertically, can you remove all the `margin-top` calculations you're currently doing?

Reply:
> provide a `margin-right` and a `gap` for .controls and...

I don't think so, because the `margin-right` would be different depending on whether `prevent-dismiss` was set (i.e. whether the dismiss button is present or not). Am I misunderstanding?

> if you set the height to be 32px (or the appropriate token) on `.controls` and then center the contents vertically...

Yes, that does work.

Reply:
> > provide a margin-right and a gap for .controls and...
> 
> I don't think so, because the margin-right would be different depending on whether prevent-dismiss was set (i.e. whether the dismiss button is present or not). Am I misunderstanding?

I didn't realize that the space was always reserved for the dismiss button even if it wasn't visible. But, I still think you could achieve the desired styling by using `visibility: hidden` to style the dismiss button when it shouldn't be shown (or alternatively, you could keep the `.dismiss` div, give it a fixed width, and have the template conditionally add the button). Either way, the goal of my suggestion was to simplify the styling so that it was easier to look at the CSS and visualize what was happening to the components. With the current styling, I think that's difficult to do, which also makes it difficul to look at the CSS and tell that it is correct.